src/python/WMCore/WMSpec/Steps/Fetchers/PileupFetcher.py

```diff
@@ -70,15 +74,21 @@ def _queryDbsAndGetPileupConfig(self, stepHelper, dbsReader, fakeSites):
             # each dataset input can generally be a list, iterate over dataset names
             blockDict = {}
```

Comment:
I don't know why you would have multiple datasets, but once you do having the dict of blocks outside of the loop seems correct to me. I'm probably missing something....